application/cache/cache.go

```diff
@@ -0,0 +1,233 @@
+				return err
+			}
+		}
+
+		// should be cached publicly. Just need to update or create an entry in the public cache,
+		// as we've already dealt with the case where visibility was originally private
+		if isPublic(request) {
+			requestAbridged, err := conversions.ConvertRequestToAPITypeAbridged(ctx, request)
+			if err != nil {
+				return err
+			}
+			requestsMap[request.UUID.String()] = requestAbridged
+			return CacheWrite(ctx, organization.Name, requestsMap)
+		}
+	}
+	return nil
+}
+
+// Gets cache entry for organization, or creates one if none exists
+// Returns true if new cache entry was created
+func getOrCreateCacheEntryPrivate(ctx context.Context, organization models.Organization) (bool, map[string]api.RequestAbridged, error) {
+	var requestsMap map[string]api.RequestAbridged
+	// if a cache value does not exist, we create it
+	if err := CacheRead(ctx, organization.Name, requestsMap); err != nil {
+		tx := models.Tx(ctx)
+
+		var org models.Organization
+		if err := org.FindByUUID(tx, organization.UUID.String()); err != nil {
+			return false, nil, err
+		}
+
+		// RequestFilterParams is currently empty because the UI is not using it
+		filter := models.RequestFilterParams{}
+
+		requests := models.Requests{}
+		if err := requests.FindByOrganization(tx, org, filter); err != nil {
+			return false, nil, err
+		}
+
+		requestsList, err := conversions.ConvertRequestsAbridged(ctx, requests)
+		if err != nil {
+			return false, nil, err
+		}
+		requestsMap := make(map[string]api.RequestAbridged)
+		for _, requestEntry := range requestsList {
+			requestsMap[requestEntry.ID.String()] = requestEntry
+		}
+
+		return true, requestsMap, CacheWrite(ctx, organization.Name, requestsMap)
+	}
+
+	return false, requestsMap, nil
+}
+
+// Gets cache entry for organization, or creates one if none exists.
+// Returns true if new cache entry was created
+func getOrCreateCacheEntryPublic(ctx context.Context) (bool, map[string]api.RequestAbridged, error) {
+	var requestsMap map[string]api.RequestAbridged
+	// if a cache value does not exist, we create it
+	if err := CacheRead(ctx, PublicRequestsKeyName, requestsMap); err != nil {
+		tx := models.Tx(ctx)
+
+		// RequestFilterParams is currently empty because the UI is not using it
+		filter := models.RequestFilterParams{}
+
+		requests := models.Requests{}
+		if err := requests.FindPublic(tx, filter); err != nil {
+			return false, nil, err
+		}
+		requestsList, err := conversions.ConvertRequestsAbridged(ctx, requests)
+		if err != nil {
+			return false, nil, err
+		}
+		requestsMap := make(map[string]api.RequestAbridged)
+		for _, requestEntry := range requestsList {
+			requestsMap[requestEntry.ID.String()] = requestEntry
+		}
+
+		return true, requestsMap, CacheWrite(ctx, PublicRequestsKeyName, requestsMap)
+	}
+	return false, requestsMap, nil
+}
+
+// we should cache non-completed public requests publicly
+func isPublic(request models.Request) bool {
```

Comment:
These three functions seem to belong in `models/request.go` as struct-attached functions on `Request`.